src/dmd/cppmangle.d

```diff
@@ -995,8 +995,19 @@ private final class CppMangleVisitor : Visitor
         // fake mangling for fields to fix https://issues.dlang.org/show_bug.cgi?id=16525
         if (!(d.storage_class & (STC.extern_ | STC.field | STC.gshared)))
         {
-            d.error("Internal Compiler Error: C++ static non-`__gshared` non-`extern` variables not supported");
-            fatal();
+            if (!target.cpp.threadLocalSupport())
```

Comment:
GCC supports `thread_local`, my comment was that you can't make any assumptions about how it is mangled in cppmangle. It is a red flag when you comment about OSX having a wrapper functions, because this is not true for GCC.

Reply:
> you can't make any assumptions about how it is mangled in cppmangle

What? Its not mangled at all. Thats why no mangling changes are necessary.

> OSX having a wrapper functions

The default environment for OSX is Apple clang which does use wrapper functions from the look of things (`_ZTW` symbols are expected which demangler.com tells me are wrapper functions). 

> because this is not true for GCC.

So what? This is DMD. That the comments don't match is irrelevant. LDC probably also works, but `target.d` is explicitly supposed to be complier specific.

Reply:
I've done some digging.  ABI isn't a concern, so I'll scratch this conversation, however TLS support is still impeded.